src/keeper.js

```diff
@@ -92,21 +93,22 @@ export default class keeper {
         let owe27 = slice18.mul(auction.price).div(decimals18);
         let tab27 = auction.tab.div(decimals18);
         // adjust covered debt to tab, such that slice better reflects amount of collateral we'd receive
-        if (owe27.gt(tab27)) {
+        if (owe27.gt(tab27) && slice18.gt(auction.lot)) {
```

Comment:
This fix is unrelated to LP tokens.  When `slice18 <= auction.lot` we were readjusting `slice18` and failing to take the auction.  This change fixes the problem and allows an array of simulations I ran to complete successfully.

Again, not sure what the original intent is here, but this entire `if/else if` block needs re-examination.

Reply:
This block of code is meant to replicate [line 364-380](https://github.com/makerdao/dss/blob/master/src/clip.sol#L364) of the corresponding solidity.  The goal is to calculate the amount of collateral which would be received from the `take` TX, rather than making the incorrect assumption that the entire `lot` would be taken at every possible auction price.

I don't see a use case where `slice` could be greater than `lot`, so I see no issue with your change.  Would appreciate a review of this block so I can address any concerns therein.